### DISCUSSION TITLE
How to monkey patch a validator onto an existing model?

### DISCUSSION THREAD

**Discussion by phdowling** [pydantic/pydantic#8260] [Question]
Hi, simple question really, I am trying to make the following work:

```python
from pydantic import BaseModel, field_validator


class Something(BaseModel):
    field_1: int
    field_2: int

    @field_validator('field_1')
    @classmethod
    def le_10_1(cls, value):
        if value > 10:
            raise ValueError(":(")
        return value


def le_10_2(cls, value):
    if value > 10:
        raise ValueError(":(")
    return value

# How to monkey-patch this validator after the class is already defined?
Something.le_10_2 = field_validator('field_2')(classmethod(le_10_2))

Something(field_1=1, field_2=1)
Something(field_1=1, field_2=11)  # this should fail, but does not as the validator is never called
Something(field_1=11, field_2=1). # this fails, as expected
```
I guess the validators are never touched again once the class is initialized, so far I haven't found a straighforward way of patching in the logic.

Creating a subclass with the validator and patching the entire class itself might be an option, but I have not gotten that to work with the particular library I am working with (it seems like it's managing to import the class before I can patch it, but that's a separate problem).

I'm using pydantic v2, version 2.5.2 to be precise.

Any help is greatly appreciated!

**Comment by spyoungtech** [comment]
If you don't have control over the code for the `Something` model before import time, then no, you can't. 

You can subclass it and add the validator though.

```py
from somelib import Something as Something_

class Something(Something_):
    @field_validator('field_2')
    @classmethod
    def le_10_2(cls, value):
        if value > 10:
            raise ValueError(":(")
        return value

Something(field_1=1, field_2=11)  # fails validation
```

However, if `Something` belongs to a library you don't have control over, that's probably not helpful if the internals of that library continue to reference `Something` rather than your subclass. If you _do_ author the library containing the `Something` class, you could formulate the class in such a way that you can change the validation behavior as needed later on, without needing a subclass.
